### DISCUSSION TITLE
Shouts, non recognized words

### DISCUSSION THREAD

**Discussion by hanz-s** [rakuri255/UltraSinger#92] [Q&A]
Hello,

is there a way to increase recognition of shouts, or voice that is'nt necessarily a word, to not get filtered when they are not recognized as a word. so one can get the timing of those shouts in the ultrastar text file? would i need a language model trained for this, oder can i add an additional model somehow which increases supports this anlong with the normal language models?

**Comment by rakuri255** [comment]
Its currently a difficult task, since the llm models only give timestamps for words. I think when you find an model on huggingface which is trained for singing, that would be the best chance.

**Comment by rakuri255** [comment]
I have been thinking about it again. Since we have the voice separated, we could analyze the volume. Then we could say that if there is no word/timestamp, then there is some kind of "lalala".
We need this approach anyway to eliminate the silent pause in the word itself.

**Reply by hanz-s** [reply]
nice! a placeholder would be worth gold. maybe one could set a param with a min volume / length of human voice to be recognized for a placeholder position.
thank your for your feedback to my question!

**Comment by rakuri255** [comment]
We have now some spectrograms to play with.

![image](https://github.com/rakuri255/UltraSinger/assets/36919102/961d39cc-1443-4667-8b08-132706567a5c)
